example/envoy-config.yaml

```diff
@@ -51,19 +53,36 @@ static_resources:
-                                "Include @demo-conf",
-                                "Include @crs-setup-demo-conf",
-                                "SecDefaultAction \"phase:3,log,auditlog,pass\"",
-                                "SecDefaultAction \"phase:4,log,auditlog,pass\"",
-                                "SecDefaultAction \"phase:5,log,auditlog,pass\"",
-                                "SecDebugLogLevel 3",
-                                "Include @owasp_crs/*.conf",
-                                "SecRule REQUEST_URI \"@streq /admin\" \"id:101,phase:1,t:lowercase,deny\" \nSecRule REQUEST_BODY \"@rx maliciouspayload\" \"id:102,phase:2,t:lowercase,deny\" \nSecRule RESPONSE_HEADERS::status \"@rx 406\" \"id:103,phase:3,t:lowercase,deny\" \nSecRule RESPONSE_BODY \"@contains responsebodycode\" \"id:104,phase:4,t:lowercase,deny\""
-                              ],
+                              "directives_map": {
+                                  "rs1": [
+                                    "Include @demo-conf",
+                                    "Include @crs-setup-demo-conf",
+                                    "SecDefaultAction \"phase:3,log,auditlog,pass\"",
+                                    "SecDefaultAction \"phase:4,log,auditlog,pass\"",
+                                    "SecDefaultAction \"phase:5,log,auditlog,pass\"",
+                                    "SecDebugLogLevel 3",
+                                    "Include @owasp_crs/*.conf",
+                                    "SecRule REQUEST_URI \"@streq /admin\" \"id:101,phase:1,t:lowercase,deny\" \nSecRule REQUEST_BODY \"@rx maliciouspayload\" \"id:102,phase:2,t:lowercase,deny\" \nSecRule RESPONSE_HEADERS::status \"@rx 406\" \"id:103,phase:3,t:lowercase,deny\" \nSecRule RESPONSE_BODY \"@contains responsebodycode\" \"id:104,phase:4,t:lowercase,deny\""
+                                  ],
+                                  "rs2": [
+                                    "Include @demo-conf",
+                                    "Include @crs-setup-demo-conf",
+                                    "SecDefaultAction \"phase:3,log,auditlog,pass\"",
+                                    "SecDefaultAction \"phase:4,log,auditlog,pass\"",
+                                    "SecDefaultAction \"phase:5,log,auditlog,pass\"",
+                                    "SecDebugLogLevel 3",
+                                    "Include @owasp_crs/*.conf",
+                                    "SecRule REQUEST_URI \"@streq /example\" \"id:101,phase:1,t:lowercase,deny\" \nSecRule REQUEST_BODY \"@rx maliciouspayload\" \"id:102,phase:2,t:lowercase,deny\" \nSecRule RESPONSE_HEADERS::status \"@rx 406\" \"id:103,phase:3,t:lowercase,deny\" \nSecRule RESPONSE_BODY \"@contains responsebodycode\" \"id:104,phase:4,t:lowercase,deny\""
+                                  ]
+                              },
+                              "default_directive": "rs1",
```

Comment:
Post merge nit. I think we should use default_directives as each key points to a set of directives (each Include or SecX is a directive)